pkg/flow/internal/controller/loader.go

```diff
@@ -445,36 +445,6 @@ func (l *Loader) populateComponentNodes(g *dag.Graph, componentBlocks []*ast.Blo
 				continue
 			}
 
-			if registration.Singleton && block.Label != "" {
-				diags.Add(diag.Diagnostic{
-					Severity: diag.SeverityLevelError,
-					Message:  fmt.Sprintf("Component %q does not support labels", componentName),
-					StartPos: block.LabelPos.Position(),
-					EndPos:   block.LabelPos.Add(len(block.Label) + 1).Position(),
-				})
-				continue
-			}
-
-			if !registration.Singleton && block.Label == "" {
-				diags.Add(diag.Diagnostic{
-					Severity: diag.SeverityLevelError,
-					Message:  fmt.Sprintf("Component %q must have a label", componentName),
-					StartPos: block.NamePos.Position(),
-					EndPos:   block.NamePos.Add(len(componentName) - 1).Position(),
-				})
-				continue
-			}
```

Comment:
@mattdurham I think we should keep this check; all components must have labels now.

Reply:
How would you feel about adding a carve out for unix exporter here? If this rolls out its going to be a huge breaking change.

Reply:
Personally, I'd rather make the change now so we're consistent across the board. We have a few more opportunities to do breaking changes before we need to settle on things for 1.0, at which point we're stuck for a while, so now's the time to make these kinds of changes.

If you're uneasy about this, we should probably have a discussion between the maintainers to figure out what we're going to do.

Reply:
Followup going to leave the  breaking change in.